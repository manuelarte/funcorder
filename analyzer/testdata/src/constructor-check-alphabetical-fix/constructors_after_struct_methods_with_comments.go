package constructor_check_alphabetical_fix

type ConstructorAfterStructMethodsWithComments struct {
	Name string
}

// GetName returns the name.
func (m ConstructorAfterStructMethodsWithComments) GetName() string {
	return m.Name
}

// SetName sets the name
// multi line comment
func (m *ConstructorAfterStructMethodsWithComments) SetName(name string) {
	m.Name = name
}

// NewOtherConstructorAfterStructMethodsWithComments This constructor creates the
// struct ConstructorAfterStructMethodsWithComments
// with a named return
func NewOtherConstructorAfterStructMethodsWithComments() (m *ConstructorAfterStructMethodsWithComments) { // want `constructor \"NewOtherConstructorAfterStructMethodsWithComments\" for struct \"ConstructorAfterStructMethodsWithComments\" should be placed before struct method \"GetName\"`
	m = &ConstructorAfterStructMethodsWithComments{Name: "John"}
	return
}

func NewConstructorAfterStructMethodsWithComments() *ConstructorAfterStructMethodsWithComments { // want `constructor \"NewConstructorAfterStructMethodsWithComments\" for struct \"ConstructorAfterStructMethodsWithComments\" should be placed before struct method \"GetName\"` `constructor \"NewConstructorAfterStructMethodsWithComments\" for struct \"ConstructorAfterStructMethodsWithComments\" should be placed before constructor \"NewOtherConstructorAfterStructMethodsWithComments\"`
	return &ConstructorAfterStructMethodsWithComments{Name: "John"}
}
