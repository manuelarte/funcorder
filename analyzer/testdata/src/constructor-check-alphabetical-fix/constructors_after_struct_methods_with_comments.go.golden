package constructor_check_alphabetical_fix

type ConstructorAfterStructMethodsWithComments struct {
	Name string
}

func NewConstructorAfterStructMethodsWithComments() *ConstructorAfterStructMethodsWithComments {
	return &ConstructorAfterStructMethodsWithComments{Name: "John"}
}

// NewOtherConstructorAfterStructMethodsWithComments This constructor creates the
// struct ConstructorAfterStructMethodsWithComments
// with a named return
func NewOtherConstructorAfterStructMethodsWithComments() (m *ConstructorAfterStructMethodsWithComments) {
	m = &ConstructorAfterStructMethodsWithComments{Name: "John"}
	return
}

// GetName returns the name.
func (m ConstructorAfterStructMethodsWithComments) GetName() string {
	return m.Name
}

// SetName sets the name
// multi line comment
func (m *ConstructorAfterStructMethodsWithComments) SetName(name string) {
	m.Name = name
}